test/evp_extra_test2.c

```diff
@@ -1326,6 +1326,24 @@ static int test_evp_pbe_alg_add(void)
 }
 #endif
 
+/*
+ * Currently, EVP_<OBJ>_fetch doesn't support
+ * colon separated alternative names for lookup
+ * so add a test here to ensure that when one is provided
+ * libcrypto returns an error
+ */
+static int evp_test_name_parsing(void)
+{
+    EVP_MD *md;
+
+    if (TEST_ptr(md = EVP_MD_fetch(mainctx, "SHA256:BogusName", NULL))) {
```

Comment:
This will show up in the test log as failed TEST_ptr() call although the whole test case is reported as passing.

It needs to be changed to !TEST_ptr_null(). It is a negative test case in the sense that it tests that EVP_MD_fetch correctly fails (i.e., returns NULL) in some circumstances.